extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetKeyValuePartitionedLayout.java

```diff
@@ -3,44 +3,93 @@
+ *           same rules.</li>
+ *           <li>Column names will be legalized via {@link NameValidator#legalizeColumnName(String, Set)
+ *           NameValidator.legalizeColumnName}.</li>
+ *           </ul>
  */
-public class ParquetKeyValuePartitionedLayout extends KeyValuePartitionLayout<ParquetTableLocationKey> {
+public class ParquetKeyValuePartitionedLayout extends URIListKeyValuePartitionLayout<ParquetTableLocationKey> {
 
     public ParquetKeyValuePartitionedLayout(
             @NotNull final File tableRootDirectory,
             @NotNull final TableDefinition tableDefinition,
             @NotNull final ParquetInstructions readInstructions) {
+        this(convertToURI(tableRootDirectory, true), tableDefinition, readInstructions);
+    }
+
+    public ParquetKeyValuePartitionedLayout(
+            @NotNull final URI tableRootDirectory,
+            @NotNull final TableDefinition tableDefinition,
+            @NotNull final ParquetInstructions readInstructions) {
         super(tableRootDirectory,
-                ParquetFileHelper::fileNameMatches,
+                ParquetFileHelper::isNonHiddenParquetURI,
                 () -> new LocationTableBuilderDefinition(tableDefinition),
-                (path, partitions) -> new ParquetTableLocationKey(path.toFile(), 0, partitions, readInstructions),
+                (uri, partitions) -> new ParquetTableLocationKey(uri, 0, partitions, readInstructions),
                 Math.toIntExact(tableDefinition.getColumnStream().filter(ColumnDefinition::isPartitioning).count()));
+        initListSupplier(readInstructions);
     }
 
     public ParquetKeyValuePartitionedLayout(
             @NotNull final File tableRootDirectory,
             final int maxPartitioningLevels,
             @NotNull final ParquetInstructions readInstructions) {
+        this(convertToURI(tableRootDirectory, true), maxPartitioningLevels, readInstructions);
+    }
+
+    public ParquetKeyValuePartitionedLayout(
+            @NotNull final URI tableRootDirectory,
+            final int maxPartitioningLevels,
+            @NotNull final ParquetInstructions readInstructions) {
         super(tableRootDirectory,
-                ParquetFileHelper::fileNameMatches,
+                ParquetFileHelper::isNonHiddenParquetURI,
                 () -> new LocationTableBuilderCsv(tableRootDirectory),
-                (path, partitions) -> new ParquetTableLocationKey(path.toFile(), 0, partitions, readInstructions),
+                (uri, partitions) -> new ParquetTableLocationKey(uri, 0, partitions, readInstructions),
                 maxPartitioningLevels);
+        initListSupplier(readInstructions);
+    }
+
+    public String toString() {
+        return ParquetKeyValuePartitionedLayout.class.getSimpleName() + '[' + tableRootDirectory + ']';
+    }
+
+    private void initListSupplier(@NotNull final ParquetInstructions readInstructions) {
```

Comment:
I really want opinion on how to make this part of the code more clear.

Reply:
Agreed. This pattern seems suspicious. I would try to see if `URIListKeyValuePartitionLayout` could take a `SeekableChannelsProvider provider` in the constructor and be explicit about using it in findKeys.